### DISCUSSION TITLE
LogicException when adding new model to an eloquent collection

### DISCUSSION THREAD

**Discussion by artworkad** [livewire/livewire#3037] [7. Bugs (archived)]
### Description

Getting LogicException **Queueing collections with multiple model connections is not supported** when adding to a collection of type Illuminate\Database\Eloquent\Collection.

### Exact steps to reproduce

Go to https://laravelplayground.com/#/snippets/ffba6077-605a-463f-a727-55ac092565ae and click the add button twice.

### Stripped-down, copy-pastable code snippets

```
class LivewireComponent extends Component
{
    public $items;

    public function mount()
    {
        $this->items = Item::get();
    }
  
    public function add()
    {
        $this->items->push(Item::make(['name' => 'Foo']));
    }

    public function render()
    {
        return view('livewire-component');
    }
}
```

```
<div>
  @foreach ($items as $index => $item)
    <p>
      The name is: {{ $item['name'] }}
    </p>
  @endforeach
  <button wire:click="add">Add</button>
</div>
```

Full code to reproduce the error: https://laravelplayground.com/#/snippets/ffba6077-605a-463f-a727-55ac092565ae

**Context**
- Livewire version: 2.4.3
- Laravel version: 8.5
- Alpine version: 2.8.2
- Browser: Chrome

**Comment by MarkusPayne** [comment]
I think the issue is with the type of collection used.  If you convert $this->items from an Illuminate\Database\Eloquent\Collection to an Illuminate\Support\Collection by changing

$this->items = Item::get();
to 
$this->items = collect(Item::get());

It works as expected example: https://laravelplayground.com/#/snippets/360a9231-6ec8-4fb7-b010-f2515c7f58e0

I can't tell you exactly why one works and the other doesn't but I faced a similar issue and this was my workaround.

**Comment by howellmartinez** [comment]
> I think the issue is with the type of collection used. If you convert $this->items from an Illuminate\Database\Eloquent\Collection to an Illuminate\Support\Collection by changing
> 
> $this->items = Item::get();
> to
> $this->items = collect(Item::get());
> 
> It works as expected example: https://laravelplayground.com/#/snippets/360a9231-6ec8-4fb7-b010-f2515c7f58e0
> 
> I can't tell you exactly why one works and the other doesn't but I faced a similar issue and this was my workaround.

Yes you can use Illuminate\Support\Collection as a workaround. However, when accessed in the Livewire component, the Models inside the collection get converted to arrays of attributes and you lose the convenience of being able to work with a collection of Models.

**Comment by jonneroelofs** [comment]
If you put $connection = 'mysql' your problems will go away :).

The problem is that when you add the second item to the collection, the first item is by that time re-hydrated and will have a connection of mysql (or whatever driver you are using). However the second item will have a connection of null. So now you have a collection with 2 models with different db connections, which causes the exception.

With my workaround the first model will also have mysql as its connection thus fixing the problem.

Hope this helps you.

**Comment by jonneroelofs** [comment]
Here is a working playground:
https://laravelplayground.com/#/snippets/764a39c0-8e5f-499d-bd2d-c3d89e4c27f6

Although there the connection has to be 'sqlite'
